src/structures/paging/mapper/mapped_page_table.rs

```diff
@@ -262,6 +309,47 @@ impl<'a, P: PhysToVirt> Mapper<Size2MiB> for MappedPageTable<'a, P> {
         Ok(MapperFlush::new(page))
     }
 
+    unsafe fn set_flags_p4_entry(
+        &mut self,
+        page: Page<Size2MiB>,
+        flags: PageTableFlags,
+    ) -> Result<MapperFlush<Size2MiB>, FlagUpdateError> {
+        let p4 = &mut self.level_4_table;
+        let p4_entry = &mut p4[page.p4_index()];
+        if p4_entry.is_unused() {
+            return Err(FlagUpdateError::PageNotMapped);
+        }
+        p4_entry.set_flags(flags);
+        Ok(MapperFlush::new(page))
```

Comment:
Flushing the `MapperFlash` will only reload the TLB entry for a single 2MiB page, which is not sufficient because there might be other 2MiB or 4KiB pages under the same level 4 entry. I'm not sure about the best solution for this. My only idea at the moment is to replace the `MapperFlush` return type with a new `MapperFlushAll` type that flushes the whole TLB.

(There are many similar places in the diff that have the same problem, but creating a separate comment for each place would be too much noise.)

Reply:
```rust
    fn flush_tlb_global() {
        use crate::instructions::interrupts::without_interrupts;
        without_interrupts(|| unsafe {
            let flags = Cr4::read_raw();
            Cr4::write_raw(flags & !Cr4Flags::PAGE_GLOBAL.bits());
            Cr4::write_raw(flags);
        });
    }

    fn flush_tlb() {
        unsafe {
            let (frame, flags) = Cr3::read();
            Cr3::write(frame, flags);
        }
    }
```
Depending on the CPU feature supporting PAGE_GLOBAL the linux kernel either chooses `flush_tlb()` or `flush_tlb_global()`. How do we switch globally on runtime?

Reply:
Ah, we can create a static on first use..

Reply:
> Ah, we can create a static on first use..

ah, well, I defer that to a future PR :-)

Reply:
fixed